### ISSUE TITLE
Can't use $link property 

### ISSUE THREAD

**Issue by martinicr** [plantuml-stdlib/Azure-PlantUML#40]
Hi, when trying to put the `$link` property in several components such as AzureServiceFabric, PlantUML diagram generation fails. Can't see in the documentation of this project if this property is supported or not.

**Comment by Potherca** [comment]
Hi! I am not familiar with `$link`. Could you post an example diagram of how it is used, or a link to the relevant PlantUML documentation?

**Comment by martinicr** [comment]
Go to this [page: ](https://github.com/plantuml-stdlib/C4-PlantUML#sprites-and-other-images). Then look for the `$link` property. You will see how this property can be used in several C4 components.

**Comment by martinicr** [comment]
You right and sorry for my lack of details on this.

In regular C4 PlantUML, you can add the `$link` property to several definitions such as this one:
```
System(twitter, "Twitter", $link="https://github.com/plantuml-stdlib/C4-PlantUML")
```
So, I tried to do the same with the `AzureServiceFabric` definition, but I got an error everytime I exported the diagram. I can't see any reference to the `$link` property in the Azure-PlantUML documentation so I thought chances are this is not supported for this kind of elements.

**Comment by Potherca** [comment]
No need for apologies 👍 

You are correct, in C4 there was a change in the code to go from ordered parameters[^1] to named parameters[^2], as that provides more flexibility[^3]

That functionality has not been implemented in this project, hence the errors.

I could make a ticket to look into implementing such things here (which is a good idea in itself) but  I can't really give a timeline in which it would be implemented.

[^1]: i.e. `Foo(a,b,c)`
[^2]: i.e. `Foo($a='a', $b='b', $c='c')`
[^3]: Such making easier to support optional parameters and not requiring a fixed order parameters need to be provided in.

**Comment by Potherca** [comment]
This has been resolved in #46 and is available from the `master` branch. It has not yet been released in a version but should be part of an upcoming 2.3 release.
